### PR TITLE
Fix cache headers

### PR DESCRIPTION
**Pull request by isoos**
- #2194
- increased max-age to a week for hashed assets
- fixed header value

### CODE REVIEW

File: app/lib/frontend/handlers/misc.dart [hunk 1]
```diff
@@ -59,19 +59,17 @@ Future<shelf.Response> staticsHandler(shelf.Request request) async {
...
-        HttpHeaders.cacheControlHeader: 'max-age: ${cacheAge.inSeconds}',
-      },
-    );
+    final headers = <String, String>{
+      HttpHeaders.contentTypeHeader: staticFile.contentType,
+      HttpHeaders.contentLengthHeader: staticFile.bytes.length.toString(),
+      HttpHeaders.lastModifiedHeader: formatHttpDate(staticFile.lastModified),
+      HttpHeaders.etagHeader: staticFile.etag,
+    };
+    if (hash == staticFile.etag) {
```
Comment (jonasfj) [inline app/lib/frontend/handlers/misc.dart]: Can we do `&& hash.isNotEmpty` please :)

File: app/lib/frontend/handlers/misc.dart [hunk 2]
```diff
@@ -59,19 +59,17 @@ Future<shelf.Response> staticsHandler(shelf.Request request) async {
...
+      headers[HttpHeaders.cacheControlHeader] =
+          'max-age=${staticLongCache.inSeconds}';
```
Comment (jonasfj) [inline app/lib/frontend/handlers/misc.dart]: Let's also make it private, following the rationale that we're over HTTPS, so hopefully there is no proxies.
Either way, we should make it explicit that it's either public or private, and private just seems most safe :)

Comment (isoos) [reply app/lib/frontend/handlers/misc.dart]: Isn't private the default? Anyway, setting it won't hurt.

Comment (jonasfj) [reply app/lib/frontend/handlers/misc.dart]: Uh, maybe, I don't know :)

**Comment by jonasfj** [comment]
r+

**Pull request merged**